### ISSUE TITLE
[Bug] [UWP] BoxView Color confused in 4.6

### ISSUE THREAD

**Issue by charlesroddie** [xamarin/Xamarin.Forms#10528]
**Edit:** the original issue was that BackgroundColors on BoxViews get confused or ignored inside Grids. However BackgroundColor should be depreciated in BoxView.
The real issue is that Color in BoxView is not shown in UWP.
Workaround: https://github.com/xamarin/Xamarin.Forms/issues/10528#issuecomment-622391246

## Original post

On UWP Grid background colours get confused easily and have non-deterministic behaviour.
In latest release 4.6. Has been a bug in the last couple of prereleases too. 4.5 OK.

#### Take a grid, put an Orange BoxView on the left, and a Red one on the right. Press F5

```fsharp
type App() =
    inherit Application()
    let grid1 =
        let g = Grid(ColumnWidths = [ GridLength.Star; GridLength.Star], VerticalOptions = LayoutOptions.FillAndExpand)
        g.Children.Add(BoxView(BackgroundColor = Color.Orange),0,0)
        g.Children.Add(BoxView(BackgroundColor = Color.Red),1,0)
        g
    do  base.MainPage <- ContentPage(Content = grid1)
```

![image](https://user-images.githubusercontent.com/19760720/80757686-7078ac80-8b2c-11ea-98f2-9d298ee8acdc.png)

#### That didn't work. Let's try closing and debugging again several more times. Finally...

![image](https://user-images.githubusercontent.com/19760720/80757973-eda42180-8b2c-11ea-836f-b351d56c8db9.png)

#### Put some text with transparent background on the left. Now you can't see the orange BoxView:

```fsharp
    let grid2 =
        let g = Grid(ColumnWidths = [ GridLength.Star; GridLength.Star], VerticalOptions = LayoutOptions.FillAndExpand)
        g.Children.Add(BoxView(BackgroundColor = Color.Orange),0,0)
        g.Children.Add(BoxView(BackgroundColor = Color.Red),1,0)
        g.Children.Add(Label(Text = "Left"),0,0)
        g
```

Expected: Orange on left, red on right, with "Left" on top of the orange box. Occasionally it does this.

More often it displays all white:

![image](https://user-images.githubusercontent.com/19760720/80760643-1e865580-8b31-11ea-812b-5d7a6b3c832c.png)

Sometimes all red:

![image](https://user-images.githubusercontent.com/19760720/80760381-b899ce00-8b30-11ea-8270-b0814ed2f630.png)

Often white on the left, red on the right.

Once the colours depended on where the window is being dragged on the screen!

**Comment by hartez** [comment]
Here's my attempt at a repro; so far, it's working just fine for me.

[_10528 Repro.zip](https://github.com/xamarin/Xamarin.Forms/files/4561930/_10528.Repro.zip)

Am I missing a layout detail?

Also, weird question, but is there any chance you have a multi-monitor setup? Does this issue persist if you run the app on your primary monitor?

**Comment by MitchBomcanhao** [comment]
I've got similar problems, BoxView controls have lost their color. They take up the space, but aren't visible.

**Comment by MitchBomcanhao** [comment]
seems like there is a new renderer for boxview in 4.6.0
BoxViewBorderRenderer

maybe this is broken and not setting the background color properly.

**Comment by MitchBomcanhao** [comment]
@hartez  
I've added this renderer override to use the old renderer
[assembly: ExportRenderer(typeof(BoxView), typeof(BoxViewRenderer))]

and now my boxviews work as expected.
the new renderer is broken. that's where the issue is.

**Comment by charlesroddie** [comment]
OK https://github.com/xamarin/Xamarin.Forms/pull/7986 is the offending PR.

**Comment by MitchBomcanhao** [comment]
A little more info - my issue might not be entirely the same as @charlesroddie - my BoxView objects used the Color property, and that does not work with the new renderer. If I use the BackgroundColor property instead, then they do appear to have the expected color using the new renderer.
Charles' issue is more about the boxview background color itself being a bit inconsistent? I've not seen this issue yet as they didn't have any color at all. In the end, the issue is likely in the same area, ie the new renderer.

**Comment by charlesroddie** [comment]
BackgroundColor is very inconsistent but I was using it by mistake. Presumably it should be depreciated in favour of Color.

Given that, investigating the inconsistencies of BackgroundColor are unimportant.

I also find with @MitchBomcanhao that setting Color displays nothing.

**Comment by bmacombe** [comment]
@charlesroddie I'll take a look, when I tested that PR I don't think I did color, but I also really just copied the previous renderer and replaced the UWP control.

**Comment by bmacombe** [comment]
I suspect it has something to do with this override I needed to do in order to keep the background color inside the border

protected override void UpdateBackgroundColor()
		{
			//background color change must be handled separately
			//because the background would protrude through the border if the corners are rounded
			//as the background would be applied to the renderer's FrameworkElement
			if (Control == null)
				return;
			Color backgroundColor = Element.BackgroundColor;
			Control.Background = backgroundColor.IsDefault ? null : backgroundColor.ToBrush();
		}

**Comment by bmacombe** [comment]
I don't think the orginal UWP box view renderer was ever updated to use the Color property

**Comment by bmacombe** [comment]
I'm not able to replicate @charlesroddie original issue with BackgroundColor either.  But I'm submitting a PR to add support for the Color property

**Comment by bmacombe** [comment]
@charlesroddie @MitchBomcanhao Sorry for the problem there, I see what I missed now with the Color property, it was just directly bound to the rectangle object in the previous renderer.  Which I copied into the new renderer, but obviously doesn't apply to a Border object.

rect.SetBinding(Shape.FillProperty, new Windows.UI.Xaml.Data.Binding { Converter = new ColorConverter(), Path = new PropertyPath("Color") });

**Comment by charlesroddie** [comment]
Yes bindings are the worst. No compile-time safety.

**Comment by samhouts** [comment]
closed by #10540
